substrate/client/chain-spec/src/genesis_config_builder.rs

```diff
@@ -84,30 +84,46 @@ where
 	/// Returns a json representation of the default `RuntimeGenesisConfig` provided by the
 	/// `runtime`.
 	///
-	/// Calls [`GenesisBuilder::create_default_config`](sp_genesis_builder::GenesisBuilder::create_default_config) in the `runtime`.
+	/// Calls [`GenesisBuilder::get_preset`](sp_genesis_builder::GenesisBuilder::get_preset) in the
+	/// `runtime` with `None` argument.
 	pub fn get_default_config(&self) -> core::result::Result<Value, String> {
+		self.get_named_preset(None)
+	}
+
+	/// Returns a JSON blob representation of the builtin `GenesisConfig` identified by `id`.
+	///
+	/// Calls [`GenesisBuilder::get_preset`](sp_genesis_builder::GenesisBuilder::get_preset)
+	/// provided by the `runtime`.
+	pub fn get_named_preset(&self, id: Option<&String>) -> core::result::Result<Value, String> {
 		let mut t = BasicExternalities::new_empty();
 		let call_result = self
-			.call(&mut t, "GenesisBuilder_create_default_config", &[])
+			.call(&mut t, "GenesisBuilder_get_preset", &id.encode())
 			.map_err(|e| format!("wasm call error {e}"))?;
-		let default_config = Vec::<u8>::decode(&mut &call_result[..])
+
+		let named_preset = Option::<Vec<u8>>::decode(&mut &call_result[..])
```

Comment:
has anyone demystified why `decode` needs a `&mut`? :))) I have always known it by heart but never got a good explanation.

Reply:
(I am assuming it never mutates the original buffer, maybe it does?)

Reply:
AIRC it is kinda [read pointer](https://github.com/paritytech/parity-scale-codec/blob/ca68d2edbe3c3b1bfcb2272202c62eb80d146195/src/codec.rs#L120) into the input buffer which must be moved during read process.